server/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -299,7 +304,14 @@ public String toString() {
 
     @Override
     public RestStatus status() {
-        return isTimedOut() ? RestStatus.REQUEST_TIMEOUT : RestStatus.OK;
+        if (RestApiVersion.current() == RestApiVersion.V_7 && isTimedOut()) {
```

Comment:
This is always `false` isn't it? (In `master` at least)

Reply:
Yes. I'm not sure what the correct way to handle this. Should I add an explicit check or create 2 different PRs for the master and 7.x branches?

Reply:
See previous comment: deprecate this (and introduce the opt-in property) in both branches and then switch to the new behaviour in `master`.

Reply:
Done!